### ISSUE TITLE
After resample ,my label becomes zero tensor!

### ISSUE THREAD

**Issue by skyous779** [fepegar/torchio#1006]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Bug summary

I am sure my label is not zero tensor. But after resampling , my label becomes zero tensors
[image_label.zip](https://github.com/fepegar/torchio/files/10080601/image_label.zip)
. I have sent my data later. 
[label.nii.gz](https://github.com/fepegar/torchio/files/10080602/label.nii.gz)

### Code for reproduction

```python
import torchio as tio
import pandas as pd
import os
subject = tio.Subject(
    
        source=tio.ScalarImage('./image.dcm'),
        label=tio.LabelMap('./label.nii.gz'),
)
subject.plot()
# I am sure that my label is not zero tensor
subject['label'].data.max() # tensor(1, dtype=torch.int16)

mytransform = tio.transforms.Compose([
        tio.ToCanonical(), 
        tio.Resample('source'), #issue 
        ])
subject1 = mytransform(subject)
subject1.plot() 

subject1['label'].data.max() # tensor(0, dtype=torch.int16)
```

### Actual outcome

# subject['label'].data.max()
tensor(1, dtype=torch.int16)

# subject1['label'].data.max()
tensor(0, dtype=torch.int16)

### Expected outcome

before resample:
![image](https://user-images.githubusercontent.com/52127540/203688157-f2e5254a-e0e6-43cb-81e4-6c11c5defb83.png)

after resample:
![image](https://user-images.githubusercontent.com/52127540/203688209-c65d122d-e796-45b2-90f1-eed61dae49ae.png)

**Comment by fepegar** [comment]
Hi! It looks like your input images are not aligned. Have you tried opening
them with e.g. 3D Slicer?

**Comment by skyous779** [comment]
Well, it's sure that my input images are not aligned, so I used tio.ToCanonical() and tio.Resample('source') to make them aligned. But it's strange that after resample, my label has nothing bright. I don't know what's wrong😭

**Comment by fepegar** [comment]
The images need to be aligned _in the physical space_, not in the grid space. If you open your images with Slicer, as I asked, the structures of interest should be aligned. If they are not, that needs to be solved before using TorchIO. The tranforms you're mentioning will do the job _if_ images are aligned in the physical space. More info: https://simpleitk.readthedocs.io/en/master/fundamentalConcepts.html

**Comment by skyous779** [comment]
Thinks a lot, I found the direction, origin and spacing of label aren't aligned with my image. Now I use simpleitk change them. I open my imags with Slicer and the structures of interest has been aligned. Thank you for your prompt reply.😁
